spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -48,49 +51,55 @@
 	public static final int DEFAULT_EXECUTOR_THREADS = 4;
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForPublisher() {
+	public ExecutorProvider publisherExecutorProvider() {
 		return FixedExecutorProvider.create(
 				Executors.newScheduledThreadPool(DEFAULT_EXECUTOR_THREADS));
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForSubscriber() {
+	public ExecutorProvider subscriberExecutorProvider() {
 		return FixedExecutorProvider.create(
 				Executors.newScheduledThreadPool(DEFAULT_EXECUTOR_THREADS));
 	}
 
 	@Bean
-	@ConditionalOnClass({GcpProperties.class, GoogleCredentials.class})
+	public ChannelProvider subscriberChannelProvider(GoogleCredentials credentials) {
```

Comment:
See my comment above on `@ConditionalOnMissingBean`.

Also, on my test app, if I specify the same `@Bean`, mine is used, not this one. Is there any case in which my defined bean doesn't take precedence over this one from the library?